addons/stock/wizard/stock_picking_return.py

```diff
@@ -71,7 +71,7 @@ def _compute_moves_locations(self):
                 wizard.move_dest_exists = move_dest_exists
                 wizard.parent_location_id = wizard.picking_id.picking_type_id.warehouse_id and wizard.picking_id.picking_type_id.warehouse_id.view_location_id.id or wizard.picking_id.location_id.location_id.id
                 wizard.original_location_id = wizard.picking_id.location_id.id
-                location_id = wizard.picking_id.location_id.id
+                location_id = wizard.picking_id.picking_type_id.default_location_return_id.id or wizard.picking_id.location_id.id
```

Comment:
Additionally, the `default_location_return_id` should have the priority on the `return_picking_type_id.default_location_dest_id`